### PR TITLE
Login with mobile and notification service.

### PR DESCRIPTION
**Pull request by poshboytl**
1. User can signup and login with both email or phone number
2. New notification service for sending notification (email or sms)

### CODE REVIEW

**Comment by poshboytl** [comment]
call for review @janx @lgn21st @tomlion @hpyhacking

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -41,7 +41,13 @@ def auth_member!
   end
 
   def auth_activated!
-    redirect_to settings_path, alert: t('private.settings.index.auth-activated') unless current_user.activated?
+    unless current_user.activated?
+      if current_user.email && !current_user.email_activated
+        redirect_to settings_path, alert: t('private.settings.index.auth-activated')
+      else current_user.phone_number && !current_user.phone_number_activated
```
Comment (janx) [inline app/controllers/application_controller.rb]: `else` should be `elsif`?

Can we add some controller spec to cover this logic?

File: app/views/identities/new.html.slim [hunk 2]
```diff
@@ -1,5 +1,6 @@
 = simple_form_for @identity, :url => '/auth/identity/register' do |f|
-  = f.input :email, :input_html => {:name => 'email', :autofocus => 'true'}
+  = f.input :country, as: :country, priority: [ "China" ], :input_html => {:name => "country"}
```
Comment (hpyhacking) [inline app/views/identities/new.html.slim]: 默认的地区应该使用 `I18n.default_locale` ?

Comment (poshboytl) [inline app/views/identities/new.html.slim]: 想了一下, 建议还是不要. 以前的手机绑定默认也是中国. 
主要的问题是, 国家和locale还是不一样的. 而且 locale和国家也是不对应的.

**Pull request merged**